pkg/workloads/containerd/watcher.go

```diff
@@ -139,6 +173,24 @@ func listenForDockerEvents(reader io.ReadCloser) {
 	}
 }
 
+func processContainerEvents(containerID string, events chan dTypesEvents.Message) {
+	for {
+		select {
+		case m, ok := <-events:
```

Comment:
`for { select { case ...` can be made into a single `for m := range events{`

Reply:
Excellent point. I originally had more things to handle in there. This will be much cleaner.